src/v/kafka/server/handlers/create_topics.cc

```diff
@@ -117,9 +120,8 @@ append_topic_configs(request_context& ctx, create_topics_response& response) {
         auto cfg = ctx.metadata_cache().get_topic_cfg(
           model::topic_namespace_view{model::kafka_namespace, ct_result.name});
         if (cfg) {
-            auto config_map = from_cluster_type(cfg->properties);
-            ct_result.configs = {
-              properties_to_result_configs(std::move(config_map))};
```

Comment:
i think you need to remove `properties_to_result_configs` in this commit `kafka: switch to report full topic configs` so that the commit will compile.
```
src/v/kafka/server/handlers/create_topics.cc:96:1: error: unused function 'properties_to_result_configs' [-Werror,-Wunused-function]
```

it's probably as simple as squashing some of the removals from the next commit back into this one?

Reply:
@dotnwat @graphcareful how do you think about trading off keeping commits minimal and compilable in general and in this case? Initially, I was trying to keep commits compilable, and then [earlier in this PR](https://github.com/redpanda-data/redpanda/pull/16922#discussion_r1516410077) Rob said that he likes to keep deletions separate from `test: ...` commits, so I'm wondering which one takes priority in this case in your opinion.

Reply:
I suspect this is a case of the warning being somewhat picky. Since the function has file scope, and you removed the last uses of it, it's probably fine to remove it here to keep the commit comipilable.

Reply:
>how do you think about trading off keeping commits minimal and compilable in general and in this case?

There is no single, good answer here, just feelings, haha. But more seriously, having commits be buildable is desirable because it makes `git bisect` easier to use. Do we expect everyone to build every commit? Definitely not. It's just a best effort kind of thing.

I agree with Rob that if you are going to remove a bunch of dead code that it should be in a different commit. It's a different logical change. But this case is different. Pretty sure most of the unused warnings are contained in -Wall, but I don't fully understand when and when it doesn't catch it.

Reply:
Thanks! I guess the rule I am going to have in mind then is that it's usually better to split out dead code removal into a separate commit unless it makes the commit unbuildable.